jerry-core/jerryscript.h

```diff
@@ -402,6 +422,11 @@ jerry_value_t jerry_exec_snapshot (const uint32_t *snapshot_p, size_t snapshot_s
 size_t jerry_parse_and_save_literals (const jerry_char_t *source_p, size_t source_size, bool is_strict,
                                       uint32_t *buffer_p, size_t buffer_size, bool is_c_format);
 
+/*
+ * External context functions.
+ */
+jerry_instance_t *jerry_create_instance (uint32_t heap_size, jerry_instance_alloc_t alloc);
```

Comment:
I am not a fond of passing callbacks to everything. Makes the code so complicated. Anyway at least add a user pointer, which is passed to the alloc function.